src/grammar/expressions.rs

```diff
@@ -16,11 +16,19 @@ use crate::parser::Parser;
 use crate::syntax::SyntaxKind;
 use crate::ParseError;
 
-pub(crate) fn parse_expr(p: &mut Parser) {
-    expr_bp(p, 0);
+/// Attempts to parse an expression if applicable
+pub(crate) fn opt_expr(p: &mut Parser) -> bool {
+    expr_bp(p, 0).is_ok()
 }
 
-fn expr_bp(p: &mut Parser, min_bp: u8) {
+/// Parses an expression
+pub(crate) fn expr(p: &mut Parser) {
+    if let Err(err) = expr_bp(p, 0) {
+        p.error(err);
+    }
+}
+
+fn expr_bp(p: &mut Parser, min_bp: u8) -> Result<(), ParseError> {
```

Comment:
May I suggest to rename parameter `min_bp`, it's not necessarily clear what it is used for.

Reply:
I added a description of binding power. Would you still like to change the name to something like `bind_pow` / `bind_pwr`?